Automattic/wp-calypso
# Plans 2023: Remove usePlanPrices abstraction and memoize getPlanPrices one

## Proposed Changes

- Removes the `usePlanPrices` hook as it's merely calling the selector underneath
- Memoizes the `getPlanPrices` selector to ensure referential integrity for same-arguments calls

Technically, both of these abstractions are unnecessary, especially since the introduction of `usePricingMetaForGridPlans`. However, the alternative (not having either) does lead to slightly more boilerplate to grab the prices (and will also require a similar but local selector in `usePricingMetaForGridPlans`). 

The cost for keeping the selector is memoization to ensure same reference on consecutive/same-argument calls. This should take care of some of the Redux warnings that pop up in the console. I am not sure of any other considerable gains right now from memoizing this selector, outside of referential integrity - which will allow us to optimize further transformations and components (I think memoizing this selector may actually be unnecessary for serving https://github.com/Automattic/wp-calypso/pull/82463).

## Testing Instructions



* Go to `/plans` and `/start/plans`
    * ensure pricing shows as expected
* Go to `/start/onboarding-pm` as the only consumer of `usePlanPrices`
    * choose free domain and free plan. ensure the popup shows the correct pricing info

## Pre-merge Checklist
